### ISSUE TITLE
Mod list offline, autoupdate isn't functioning

### ISSUE THREAD

**Issue by Purpzie** [Dual-Iron/RwModLoader#17]
**Describe what you were trying to do and why**
I'm simply trying to view the mod list. Also, autoupdate seems to be failing.

(For example, "Many More Fixes" is still 1.0.0 even though it's 1.0.1, mentioned here in the RW discord: https://discord.com/channels/291184728944410624/804088181904375819/807564814586216449)

**To Reproduce**
Unfortunately I'm not sure what's causing this or if it's just a local problem. Here are some logs.

- No mods enabled. Opened game, viewed mod list, and exited: [LogOutput.log](https://github.com/Dual-Iron/RwModLoader/files/8978508/LogOutput.log)

- No mods enabled. Opened game, enabled "Many More Fixes" and ConfigMachine, viewed mod list, and exited: [LogOutput.log](https://github.com/Dual-Iron/RwModLoader/files/8978537/LogOutput.log)

**Expected behavior**
Autoupdate works, the mod list is viewable.

**Actual behavior**
Autoupdate doesn't work, the mod list is always offline.

**Screenshots**
![image](https://user-images.githubusercontent.com/25022704/175606634-a91c8473-ca2a-408e-b2c6-c68d68f3a887.png)

**Additional context**
My system is configured to use Cloudflare DNS, and I've heard some websites and APIs will reject that.

**Comment by Dual-Iron** [comment]
Thanks for your report!

Trying to reproduce myself, I don't encounter any errors. However, reading through your log, there's definitely an SSL error occurring when your system tries to connect to https://rdb.dual-iron.xyz/mods. `Error while adding mods: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version`

> My system is configured to use Cloudflare DNS, and I've heard some websites and APIs will reject that.

Is there any way you could disable this to see if it works without it? I'm not sure what I can do to help, unfortunately...

**Comment by Purpzie** [comment]
Oh woops, apologies for reporting a local problem then.

It actually stopped occurring after connecting an ethernet cable to my PC. I would've tried that before making the report, but I was living in a different room without one at the time.

**Comment by Mr-Foxtrot** [comment]
Hi, I also seemingly have this issue. The mod browser refuses to load, and i've dug into the log to see the same error.
`[Error  :     Realm] Error while adding mods: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version`
I don't use Cloudflare DNS, instead I use google's DNS if that matters at all. I've tried changing DNS, disabling firewall, checking certificates, using a VPN, restarting my PC, and reinstalling Rain World (several times), etc. etc.

Bepinex Output log: [LogOutput.log](https://github.com/Dual-Iron/RwModLoader/files/9054754/LogOutput.log)

Is it perhaps an issue with the mod, or something else on my end?

**Comment by Daldwin** [comment]
Hello, I am also having the same issue, with the mod browser always offline, and the same error message in the log.
[LogOutput.log](https://github.com/Dual-Iron/RwModLoader/files/9174732/LogOutput.log)
I tried reinstalling realm, reinstalling the game, changing my DNS, VPN, firewall settings, and the windows setting to activate the various tls versions but have been the getting the exact same result every single time.

I found with a bit a digging that people have had a similar error when using git. It didn't help me much but it might help you identify the source of the error.

https://github.com/glennhickey/progressiveCactus/issues/93

https://www.ibm.com/mysupport/s/question/0D50z00005q4Ff6CAE/error1407742essl-routinesssl23getserverhellotlsv1-alert-protocol-version?language=fr
